Makefile

```diff
@@ -17,7 +16,4 @@ endif
 
 .PHONY: test
 test: set-git-ssh
-	cd test && git fetch; \
-	for g in $(shell cd test && go run scripts/test-wrapper.go); do \
-		go test -timeout 30m -race -v -run $$g; \
-	done
+	./test/dispatch-ci.sh
```

Comment:
This may be a bit out of place :thinking: above we specifically trigger based on `DISPATCH_CI` but not here? This would seem to suggest the Makefile here is only intended for dispatch and if so we should probably make that more clear.

Reply:
So, at the moment, `./test/dispatch-ch.sh` can run outside of Dispatch without any problem. But with the caveat that it exercises only a sub-set of tests, which we don't want. 

The reason for exercising only subset is given in the `./test/dispatch-ch.sh` file, and the moment, I am trying to think, how can I enhance that. 

These was my thinking:

a) Make sure that dispatch-ci.sh can run entire test-suite. ( I didn't want to call that script as test-ci.sh)
b) Then call it as `test-ci.sh` and make sure, `make test` is the canonical way of test invocation.

At the moment, we use `make test` only in dispatch, so I did not think of adding any more comments.

But open if you have any suggestions on what I could do. Adding a comment is okay with me.

Reply:
I'm confused by that last. Can I run `make test` on my local computer?

Reply:
> . Can I run make test on my local computer?
@joejulian  - you could, but it won't exercise the full suite (because of change introduced here and bug present in dispatch).

```
(base) senthil@mahalonobis:~/github/kubernetes-base-addons-clone2$ make
./test/dispatch-ci.sh
++ dirname ./test/dispatch-ci.sh
+ cd ./test
+ echo 'Setup Konvoy'
Setup Konvoy
+ source ./scripts/setup-konvoy.sh v1.4.1
++ KONVOY_VERSION=v1.4.1
++ '[' -z v1.4.1 ']'
++ KONVOY_VERSION=v1.4.1
++ set -euo pipefail
++ curl --silent https://downloads.mesosphere.io/konvoy/konvoy_v1.4.1_linux.tar.bz2 -o -
++ tar --extract -O -jf - konvoy_v1.4.1/konvoy
++ chmod +x konvoy
+ echo 'git fetch branches.'
git fetch branches.
+ git fetch
+ echo 'INFO: Exercise TestAwsGroup '
INFO: Exercise TestAwsGroup 
+ go test -tags experimental -timeout 60m -race -v -run TestAwsGroup



```

Reply:
My explanation was
a) We don't run `make test` on Teamcity, so it is safe.
b) I am not sure if developers are running `make test` locally. That's the caveat. But if they run, it will notice that only a subset is exercised.

Reply:
![image](https://user-images.githubusercontent.com/952828/80438032-0eb00c80-88b8-11ea-85ab-803b0b06344b.png)

Testing locally is our expectation and is common practice in most repos that have a Makefile and tests. Straying from that would cause confusion that I think should be avoided.

Reply:
(wow, that image got big!)

Reply:
So, should I introduce a new target for dispatch and let the`make test` run for local tests exercising the complete test suite?

Reply:
@joejulian  - I have addressed the comments here. I have made sure that `make test` stays generic, and introduced a new make target for dispatch.

Please review it again.